### PR TITLE
Guide to advanced tools for WebAssembly binary processing.

### PR DESCRIPTION
**Pull request by adanilo**
Covers WABT and Binaryen, mostly short descriptions pointing at the definitive references. Also includes added button in the header to get to the tools page.

### CODE REVIEW

File: tools/index.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+---
+## Advanced Tools
+
+WebAssembly is supported by a number of tools to help developers build and process source files and generated binary content. If you're a compiler writer, enjoy low level coding or just want to experiment with the raw WebAssembly format, these tools are for you.
+
+<div class="flash flash-warn flash-messages">
+Note, at the date of writing - Oct 2016, the WebAssembly binary format is close to final (expected to be stable) and developer feedback is encouraged before the format is locked down for version 1.0 which is expected to be early 2017. Browser previews of WebAssembly support can be used for experimentation and feedback can be sent to the <a href="https://www.w3.org/community/webassembly/">community group</a> or by posting issues on the <a href="https://github.com/WebAssembly/design">design repository</a>.
+</div>
+
+There are two distinct sets of tools that are of interest to compiler writers or developers who want to work with WebAssembly binary generated by other tools like [Emscripten](http://kripken.github.io/emscripten-site/):
```
Comment (jfbastien) [inline tools/index.md]: "there are currently"? I'd rather not make it sound like people can't do their own thing if they want to.

File: tools/index.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+*   [WABT](https://github.com/WebAssembly/wabt) - The WebAssembly Binary Toolkit
+*   [Binaryen](https://github.com/WebAssembly/binaryen) - Compiler and toolchain infrastructure
+
+### WABT: The WebAssembly Binary Toolkit
+
+This toolkit supports conversion between the binary WebAssembly format into and out of a human readable text format. The text format is a type of [S-expression](https://en.wikipedia.org/wiki/S-expression) that makes it convenient to work with the output of a WebAssembly compiler for analysis or debugging purposes, etc.
+
+Note, that the S-expression format supported by WABT is not WebAssembly itself. It's one of many possible text formats that could represent the content of a WebAssembly file, so it's been developed as a convenient format for the purposes of decode and encode by WABT tools. Developers could easily build decoder/encoder implementations for any other textual format capable of expressing the WebAssembly [Abstract Syntax Tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree) (WAST).
```
Comment (jfbastien) [inline tools/index.md]: Stack machine instead?

File: tools/index.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+#### wast2wasm tool
+
+This command line tool performs the inverse of **wasm2wast**, i.e. it converts the S-expression WAST file into a binary WebAssembly file.
+
+Using **wasm2wast** and **wast2wasm** together allows lossless round tripping of WebAssembly binaries, and provides developers with a convenient way to manipulate the content of WebAssembly binaries using external tools.
+
+#### wasm-interp tool
+
+This is an interpreter that lets developers run a WebAssembly binary from the command line stand-alone. It implements a stack-machine based interpreter that interprets the WebAssembly binary directly. This differs from how a browser would JIT the WebAssembly binary into native code for it's target architecture at load time.
```
Comment (jfbastien) [inline tools/index.md]: s/it's/its/

File: tools/index.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+### Binaryen
+
+[Binaryen](https://github.com/WebAssembly/binaryen) is a comprehensive set of tools with supporting infrastructure for use as the backend of compilers that are targeting WebAssembly as an output format. It has a C API and implements its own internal intermediate representation ([IR](https://en.wikipedia.org/wiki/Intermediate_representation)) of program logic and can perform a number of optimizations on the IR, support parallelization of code generation, etc.
+
+For example, binaryen is used as part of the compiler **[asm2wasm](https://github.com/WebAssembly/binaryen/blob/master/src/asm2wasm.h)** that can convert asm.js files into WebAssembly files. It's also used for supporting the [LLVM](http://llvm.org/) compiler infrastructure generation of WebAssembly and compilation from [Rust](https://www.rust-lang.org/en-US/).
+
+Developers working on compilers, advanced optimization techniques and so forth should take advantage of binaryen and it's tools that include a shell that can load and interpret WebAssembly code, assembler and disassembler, converters for asm.js and LLVM .s files into WebAssembly and so on.
```
Comment (jfbastien) [inline tools/index.md]: s/it's/its/

**Comment by kripken** [comment]
Nice, lgtm (with @jfbastien's comments).

**Comment by s3ththompson** [comment]
LGTM, will fix these nits in my merge giving the timing

**Comment by s3ththompson** [comment]
Closed by 1e9ec0759b7ed80510714e72018dc7b61c34e731

**Pull request closed without merging**